example_with_variants_test.go

```diff
@@ -0,0 +1,35 @@
+package unleash_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/Unleash/unleash-client-go/v3"
+)
+
+func Test_withVariants(t *testing.T) {
+	demoReader, err := os.Open("demo_app_toggles.json")
+	if err != nil {
+		t.Fail()
+	}
+	err = unleash.Initialize(
+		unleash.WithListener(&unleash.DebugListener{}),
+		unleash.WithAppName("my-application"),
+		unleash.WithRefreshInterval(5*time.Second),
+		unleash.WithMetricsInterval(5*time.Second),
+		unleash.WithStorage(&unleash.BootstrapStorage{Reader: demoReader}),
+		unleash.WithUrl("https://localhost:4242"),
+	)
+
+	if err != nil {
+		t.Fail()
+	}
+
+	enabled := unleash.GetVariant("ActivityLog")
+	fmt.Printf("feature is enabled? %v\n", enabled.Enabled)
+	if enabled.Enabled == true {
+		t.Fatalf("Expected feature to be enabled")
+	}
```

Comment:
I believe that according to the spec `enabled.Enabled` should be true, but it's actually false:
```json
{
      "name": "ActivityLog",
      "type": "release",
      "enabled": true,
      "stale": false,
      "strategies": [
        {
          "name": "Environment",
          "parameters": {
            "Dev": "",
            "QA": "",
            "Test": "",
            "Training": "",
            "Prod": ""
          },
          "constraints": []
        }
      ],
      "variants": [
        {
          "name": "type_1",
          "weight": 500,
          "payload": { "type": "string", "value": "opt_1" },
          "overrides": [],
          "weightType": "variable",
          "stickiness": "default"
        },
        {
          "name": "type_2",
          "weight": 500,
          "payload": { "type": "string", "value": "opt_2" },
          "overrides": [],
          "weightType": "variable",
          "stickiness": "default"
        }
      ]
    },
```

Reply:
This one was not acting as we expected because of the strategy named `Environment` that doesn't exist in the default Unleash configuration. We solved this by switching to another feature toggle using the `default` strategy